sdk/appconfiguration/app-configuration/test/helpers.spec.ts

```diff
@@ -1,36 +1,169 @@
+  
+      assert.equal(
+        "'etagishere'",
+        quoteETag("'etagishere'")
+      );
+  
+      assert.equal(
+        "*",
+        quoteETag("*")
+      );
+    });
+  });
+  
+  describe("formatWildcards", () => {
+    it("undefined", () => {
+      const result = formatWildcards({
+        keys: undefined,
+        labels: undefined
+      });
+  
+      assert.ok(!result.key);
+      assert.ok(!result.label);
+    });
+  
+    it("single values only", () => {
+      const result = formatWildcards({
+        keys: ["key1"],
+        labels: ["label1"]
+      });
+  
+      assert.equal("key1", result.key);
+      assert.equal("label1", result.label);
+    });
+  
+    it("multiple values", () => {
+      const result = formatWildcards({
+        keys: ["key1", "key2"],
+        labels: ["label1", "label2"]
+      });
+  
+      assert.equal("key1,key2", result.key);
+      assert.equal("label1,label2", result.label);
+    });
   });
-})
\ No newline at end of file
+  
+  describe("extractAfterTokenFromNextLink", () => {
+    it("token is extracted and properly unescaped", () => {
+      let token = extractAfterTokenFromNextLink(
+        "/kv?key=someKey&api-version=1.0&after=bGlah%3D"
+      );
+      assert.equal("bGlah=", token);
+    });
+  });
+
+  it("makeConfigurationSettingsFieldsThrow", () => {
+    const response: ConfigurationSetting & HttpResponseField<any> & HttpResponseFields = {
+      key: "undefined",
+      _response: {
+        request: {
+          url: "unused",
+          abortSignal: {
+            aborted: true,
+            addEventListener: () => { },
+            removeEventListener: () => { }
+          },
+          method: "GET",
+          withCredentials: false,
+          headers: new HttpHeaders(),
+          timeout: 0,
+          clone: function () { return this; },
+          validateRequestProperties: () => { },
+          prepare: function (options) { return this; },
+        },
+        status: 204,
+        headers: new HttpHeaders(),
+        bodyAsText: "",
+        parsedHeaders: {}
+      },
+      statusCode: 204
+    };
+
+    makeConfigurationSettingsFieldsThrow(response,
+      "My error message",
+      "My error code");
+    
+    for (const name of getAllConfigurationSettingFields()) {
+      assert.throws(() => (<any>response)[name], (err: ResponseBodyNotFoundError) => {
+        assert.equal("ResponseBodyNotFoundError", err.name);
+        assert.equal("My error message", err.message);
+        assert.equal("My error code", err.code);
+        assert.strictEqual(response._response, err.response);
+        return true;
+      });
+    }
+
+    // These point is these properties are untouched and won't throw
+    // since they're the only properties the user is allowed to touch on these 
+    // "body empty" objects.
+    assert.equal(204, response._response.status);
+    assert.equal(204, response.statusCode);
+  });
+
+  function getAllConfigurationSettingFields() : string[] {
```

Comment:
if you make this return `key of ConfigurationSetting` I bet you could avoid the `any` cast above on line 139

Reply:
Done. I still had to cast (Object.keys -> (keyof ConfigurationSetting)[]) but at least it was within getAllConfigurationSettingFields, which does seem nicer.